### PR TITLE
Add SDL tools to website deployment pipeline

### LINKED ISSUE #9304: Enable SDL on website deployment pipeline

### PR DESCRIPTION
**Pull request by tylerbutler**
Fixes #9304.

### CODE REVIEW

File: tools/pipelines/build-docs.yml [hunk 1]
```diff
@@ -118,14 +138,16 @@ stages:
...
+                uploadAsLatest: ${{ variables.isMain }}
 
 - stage: site_build
   displayName: 'Build website'
   dependsOn: json
   jobs:
     - job: build_site
       condition: in(stageDependencies.json.upload_json.result, 'Succeeded', 'SucceededWithIssues')
+      pool:
+        vmImage: windows-latest
```
Comment (tylerbutler) [inline tools/pipelines/build-docs.yml:150]: SDL tools work better on windows at the moment.

File: docs/.gdnbaselines [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      "alternativeSignatures": [
+        "b5f214e29b5462952a2841914ef8d1fc8141a9e8a69ea2b2be6c0cd409bf15d7"
+      ],
+      "target": "docs/themes/thxvscode/assets/js/bundle.js",
+      "memberOf": [
+        "default"
+      ],
+      "tool": "eslint",
+      "ruleId": "@microsoft/sdl/no-inner-html",
+      "justification": "Comes from jquery dependency. Tracked by https://github.com/microsoft/FluidFramework/issues/9316.",
```
Comment (tylerbutler) [inline docs/.gdnbaselines:22]: This file defines the "baseline errors" that we have right now and tracks that they get resolved by the expiration date.

File: tools/pipelines/build-docs.yml [hunk 3]
```diff
@@ -145,6 +167,56 @@ stages:
...
+      - task: UseDotNet@2
+        displayName: 'Use .NET Core 3.x'
+        condition: succeededOrFailed()
+        inputs:
+          packageType: sdk
+          version: 3.x
+
+      - task: securedevelopmentteam.vss-secure-development-tools.build-task-eslint.ESLint@1
+        condition: succeededOrFailed()
+        displayName: 'Run ESLint'
```
Comment (heliocliu) [inline tools/pipelines/build-docs.yml:180]:
```suggestion
        displayName: 'Run ESLint'
        condition: succeededOrFailed()
```

File: tools/pipelines/build-docs.yml [hunk 4]
```diff
@@ -145,6 +167,56 @@ stages:
...
+
+      - task: securedevelopmentteam.vss-secure-development-tools.build-task-publishsecurityanalysislogs.PublishSecurityAnalysisLogs@3
+        displayName: 'Publish Guardian Artifacts - All Tools'
+        inputs:
+          ArtifactType: M365
+        condition: succeededOrFailed()
```
Comment (heliocliu) [inline tools/pipelines/build-docs.yml:186]:
```suggestion
        displayName: 'Publish Guardian Artifacts - All Tools'
        condition: succeededOrFailed()
        inputs:
          ArtifactType: M365
```

File: tools/pipelines/build-docs.yml [hunk 5]
```diff
@@ -145,6 +167,56 @@ stages:
...
+        inputs:
+          ArrowServiceConnection: 'Arrow_FluidFramework_internal'
+          AssetGroupName: 'fluidframework_$(System.TeamProject)_$(Build.DefinitionName)'
+          AssetNumber: '$(Build.BuildId)'
+          IsShipped: false # based on value of arrow.releasedtoproduction variable
+          DropsToRetain: 'CodeAnalysisLogs'
+
+      - task: securedevelopmentteam.vss-secure-development-tools.build-task-postanalysis.PostAnalysis@2
+        condition: succeededOrFailed()
+        displayName: 'Guardian Break'
```
Comment (heliocliu) [inline tools/pipelines/build-docs.yml:200]:
```suggestion
        displayName: 'Guardian Break'
        condition: succeededOrFailed()
```

File: docs/.gdnbaselines [hunk 6]
```diff
@@ -0,0 +1,44 @@
+{
```
Comment (heliocliu) [inline docs/.gdnbaselines:1]: how does this file work?

**Review by heliocliu — APPROVED** [review]

**Pull request merged**